ff/serving/metadata/metadata_test.go

```diff
@@ -0,0 +1,1191 @@
+		t.Fatalf("Failed to set up metadata server and create  %s", err)
+	}
+	streamModel := makeStreamMockModel()
+	if err := streamModel.SendFromClient(&pb.Name{
+		Name: "user_fraud_random_forest",
+	}); err != nil {
+		t.Fatalf("Failed to send from client 'Models'%s", err)
+	}
+	go func() {
+		if err := metadataServer.GetModels(streamModel); err != nil {
+			t.Fatalf("Failed to get models%s", err)
+		}
+	}()
+	go func() {
+		var v pb.Empty
+		if err := metadataServer.ListModels(&v, streamModel); err != nil {
+			t.Fatalf("Failed to get models%s", err)
+		}
+	}()
+	_, errRecieveModel := streamModel.RecvToClient()
+	if errRecieveModel != nil {
+		t.Fatalf("failed to recieve model")
+	}
+}
+
+func TestFeaturesMethodsGetList(t *testing.T) {
+	metadataServer, err := initializeCreate()
+	if err != nil {
+		t.Fatalf("Failed to set up metadata server and create  %s", err)
+	}
+	streamfeatures := makeStreamMockFeaturesNonEmpty()
+	if err := streamfeatures.SendFromClient(&pb.Name{
+		Name: "user_credit_score",
+	}); err != nil {
+		t.Fatalf("Failed to send from client 'Features'%s", err)
+	}
+	go func() {
+		if err := metadataServer.GetFeatures(streamfeatures); err != nil {
+			t.Fatalf("Failed to get feature%s", err)
+		}
+	}()
+	go func() {
+		var v pb.Empty
+		if err := metadataServer.ListFeatures(&v, streamfeatures); err != nil {
+			t.Fatalf("Failed to get features%s", err)
+		}
+	}()
+	_, errRecieveFeature := streamfeatures.RecvToClient()
+	if errRecieveFeature != nil {
+		t.Fatalf("failed to recieve features")
+	}
+}
+
+func TestFeatureVarAndTrainingSetMethods(t *testing.T) {
+	metadataServer, err := initializeCreate()
+	if err != nil {
+		t.Fatalf("Failed to set up metadata server and create  %s", err)
+	}
+	streamFeatureVariant := makeStreamMockFeaturesVar()
+	if err := streamFeatureVariant.SendFromClient(&pb.NameVariant{
+		Name:    "user_credit_score",
+		Variant: "default",
+	}); err != nil {
+		t.Fatalf("Failed to send from client 'feature' var%s", err)
+	}
+	go func() {
+		if err := metadataServer.GetFeatureVariants(streamFeatureVariant); err != nil {
+			t.Fatalf("Failed to get feature var%s", err)
+		}
+	}()
+	streamTrainingSetVar := makeStreamMockTrainingSetVar()
+	if err := streamTrainingSetVar.SendFromClient(&pb.NameVariant{
+		Name:    "is_fraud",
+		Variant: "default",
+	}); err != nil {
+		t.Fatalf("Failed to send from client 'TrainingsetVariant'%s", err)
+	}
+	go func() {
+		if err := metadataServer.GetTrainingSetVariants(streamTrainingSetVar); err != nil {
+			t.Fatalf("Failed to get training set variants%s", err)
+		}
+	}()
+	_, errRecieveTrainingSetVar := streamTrainingSetVar.RecvToClient()
+	if errRecieveTrainingSetVar != nil {
+		t.Fatalf("failed to recieve")
+	}
+	featureVariant, errRecieveFeatureVariant := streamFeatureVariant.RecvToClient()
+	if errRecieveFeatureVariant != nil {
+		t.Fatalf("failed to recieve feature var")
+	}
+	var containsTrainingset bool = false
+	for _, feature := range featureVariant.Trainingsets {
```

Comment:
this variable should be called ts, trainingSet, etc. was really confused

Reply:
This is only testing training set, there are a lot of other resources that get things updated